p2p/chain.py

```diff
@@ -720,6 +610,138 @@ class RegularChainSyncer(FastChainSyncer):
         return head.block_number
 
 
+class PeerRequestHandler:
+
+    def __init__(self, db: 'AsyncHeaderDB', logger: TraceLogger, token: CancelToken) -> None:
+        self.db = db
+        self.logger = logger
+        self.cancel_token = token
+
+    async def handle_get_block_bodies(self, peer: ETHPeer, block_hashes: List[Hash32]) -> None:
+        chaindb = cast('AsyncChainDB', self.db)
+        bodies = []
+        # Only serve up to eth.MAX_BODIES_FETCH items in every request.
+        for block_hash in block_hashes[:eth.MAX_BODIES_FETCH]:
```

Comment:
Same thought here about moving the limits up a level to the call site of this function.

```python
bodies = self._handler.handle_get_block_bodies(peer, requested_block_hashes[:MAX_BODIES_FETCH])
```